### ISSUE TITLE
Zypper always reports changed

### ISSUE THREAD

**Issue by jcrisologo** [ansible/ansible-modules-extras#299]
Issue Type:

Bug Report

Ansible Version:

1.8.2

Environment:

RHEL 7

Summary:

The zypper module installs packages by running zypper in a shell.  It then checks the return code to determine if anything was changed.  However, zypper will return success (0) even when it reports "Nothing to do."

Steps To Reproduce:
1. Install a package with zypper

```
  ansible -m zypper -a 'name=ssh state=present' -i inventory sles_host
```
1. Do it again

```
  ansible -m zypper -a 'name=ssh state=present' -i inventory sles_host
```

Expected Results:

Ansible may report that a change occurred after the first step.
Ansible will report that no change occurred after the second step.

Actual Results:

Ansible reports that a change occurred after both steps.

**Comment by jcrisologo** [comment]
Would simply inspecting the stdout output be a reasonable solution?

**Comment by robinro** [comment]
rc=0 on "Nothing to do." is correct, but when the package is already installed, then ansible won't even call zypper. At least here (ansible 1.9.1 with openSuSE 13.2) I get changed=False on the second run.

Maybe there was a fix after 1.8.2, but I use ansible far longer than that and it always worked for me.

**Comment by robinro** [comment]
This issue is old and I can't reproduce it.
close_me
